packages/assets-controllers/src/TokenListController.ts

```diff
@@ -140,8 +151,20 @@ export class TokenListController extends BaseControllerV2<
     this.intervalDelay = interval;
     this.cacheRefreshThreshold = cacheRefreshThreshold;
     this.chainId = chainId;
+    this.selectedAddress = selectedAddress;
     this.updatePreventPollingOnNetworkRestart(preventPollingOnNetworkRestart);
     this.abortController = new WhatwgAbortController();
+    if (onPreferencesStateChange) {
+      onPreferencesStateChange(({ selectedAddress: newSelectedAddress }) => {
+        const currentSelectedAddress = this.selectedAddress;
```

Comment:
When the selected address changes, should `this.selectedAddress` also be updated? It seems like `this.selectedAddress` is only set once when this controller is instantiated, but never changed, so if you change the selected address back so that it matches `this.selectedAddress`, then `this.fetchTokenList()` won't be called.